src/hotspot/share/opto/node.hpp

```diff
@@ -1712,6 +1729,33 @@ class Unique_Node_List : public Node_List {
 #endif
 };
 
+// Definition must appear after complete type definition of Unique_Node_List
+template <typename Callback, typename Check>
+void Node::visit_uses(Callback callback, Check is_boundary) {
+  ResourceMark rm;
+  // Unique_Node_List guarantees no duplicates in the worklist. Note that we
+  // never pop anything from the worklist, since that could result in applying
+  // the callback for a use more than once (if it is at some point readded to
+  // the worklist).
+  Unique_Node_List worklist;
+  // The initial worklist consists of the direct uses
+  for (DUIterator_Fast kmax, k = fast_outs(kmax); k < kmax; k++) {
+    worklist.push(fast_out(k));
+  }
```

Comment:
An alternative could be to just push "this", and then in the loop: `if (use == this) { continue; }`
Would just save the code duplication.

Reply:
True, but I do not like having a conditional in the loop that is redundant in all but the first iteration. I prefer the current solution, but I'm also OK with changing. Maybe there is a third even more elegant solution?